app.js

```diff
@@ -0,0 +1,15 @@
+const express = require("express")
+const { retrieveTopics } = require("./controllers/topics.controllers")
+
+const app = express()
+
+app.use(express.json())
+
+app.get("/api/topics", retrieveTopics)
+
+app.use((req, res, next) => {
+        res.status(404).send({ msg: "Invalid endpoint" })
+
+})
```

Comment:
This isn't strictly how I want us to handle endpoints that aren't found because eventually this error handling middleware will have a load of logic in it and the only error we want to throw if the error reaches the bottom of the middleware is 500 Internal Server Error not a 404.
 
 So we need a different way to send a 404 after all the valid endpoints have been tried but before the official error handling kicks in. This is where you might see internet examples of people using `app.all` to handle invalid endpoints. I would do some research on this and see if there are any examples that do this in the app.js